packages/manager/apps/pci-kubernetes/src/api/hooks/useSubnets.ts

```diff
@@ -0,0 +1,65 @@
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  getPrivateNetworkSubnets,
+  TPrivateNetworkSubnet,
+} from '@/api/data/subnets';
+
+export const getPrivateNetworkSubnetsQuery = (
+  projectId: string,
+  privateNetworkId: string,
+) => ({
+  queryKey: ['project', projectId, 'network', privateNetworkId, 'subnet'],
+  queryFn: (): Promise<TPrivateNetworkSubnet[]> =>
+    getPrivateNetworkSubnets(projectId, privateNetworkId),
+});
+
+export const usePrivateNetworkSubnets = (
+  projectId: string,
+  privateNetworkId: string,
+  regionName: string,
+) => {
+  const { t } = useTranslation('network-add');
+
+  const { data, isLoading, error } = useQuery({
+    ...getPrivateNetworkSubnetsQuery(projectId, privateNetworkId),
+    enabled: !!projectId && !!privateNetworkId,
+  });
+
+  const augmentSubnets = (subnets, { none = false } = {}) => {
+    const noneItem = none
+      ? {
+          id: null,
+          displayedLabel: t('kubernetes_network_form_subnet_none'),
+        }
+      : null;
```

Comment:
i don't think the noneItem should be part of the usePrivateNetworkSubnets hook

noneItem is a UI concept related to item selection in a dropdown list for example, so it would be better if this noneItem would be appended in the view instead of in the hook